### PR TITLE
chore: replace custom properties binding implementation with PropertyBindingSupport from camel support

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java [hunk 1]
```diff
@@ -331,26 +331,27 @@ protected ReactiveExecutor createReactiveExecutor() {
...
-            }
+
+        CamelContextAware.trySetCamelContext(result, context);
+
+        Properties props = getPropertiesComponent().loadProperties();
+        if (props != null && !props.isEmpty()) {
+            PropertyBindingSupport.bindProperties(
+                this,
+                result,
+                new HashMap<String, Object>((Map) props),
```
Comment (ppalaga) [inline extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java]: Is there any good reason to allocate a new `HashMap` ? Simply `(Map) props` would compile too.

File: extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelRuntime.java [hunk 2]
```diff
@@ -77,21 +79,17 @@ public void stop() throws Exception {
...
         try {
             this.context = createContext();
 
-            // Configure the camel context using properties in the form:
-            //
-            //     camel.context.${name} = ${value}
-            //
-            RuntimeSupport.bindProperties(properties, context, PFX_CAMEL_CONTEXT);
+            if (properties != null) {
+                PropertyBindingSupport.bindProperties(context, context, new HashMap<String, Object>((Map) properties), PFX_CAMEL_CONTEXT);
```
Comment (ppalaga) [inline extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelRuntime.java]: Is there any good reason to allocate a new `HashMap` ? Simply `(Map) props` would compile too.

File: extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java (hunk 1 above)
Comment (lburgazzoli) [reply extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java]: no, leftover from another solution that I had before

File: extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelRuntime.java (hunk 2 above)
Comment (lburgazzoli) [reply extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelRuntime.java]: fixed

File: extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java (hunk 1 above)
Comment (lburgazzoli) [reply extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java]: fixed

Comment (lburgazzoli) [reply extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java]: @ppalaga 

ah no sorry, now I recall the reason: `PropertyBindingSupport.bindProperties` consumes the map so it removes every element that has been bound to the target leading to unexpected results.

pushed a new version that restores hash map, I need to work on the camel side to have a better solution but in the meantime I would merge the PR as it is as `PropertyBindingSupport` to support OGNL like property setting.

**Comment by oscerd** [comment]
There is a conflict.

**Pull request merged**